static/src/javascripts/projects/common/modules/ui/autoupdate.js

```diff
@@ -25,190 +34,189 @@ define([
-        this.notification = '<';
-        this.updateDelay = options.delay;
-
-        this.template =
-            '  <button class="u-button-reset live-toggler live-toggler--autoupdate live-toggler--on js-auto-update js-auto-update--on"' +
-            '          data-action="off" data-link-name="autoupdate off" title="Turn auto update off">' +
-            '    <span class="live-toggler__label">Auto update:</span>' +
-            '    <span class="u-h">is</span>' +
-            '    <span class="rounded-icon live-toggle__value">On</span>' +
-            '    <span class="u-h">(turn off)</span>' +
-            '  </button>' +
-            '  <button class="u-button-reset live-toggler live-toggler--autoupdate live-toggler--off js-auto-update js-auto-update--off"' +
-            '          data-action="on" data-link-name="autoupdate on" title="Turn auto update on">' +
-            '    <span class="live-toggler__label">Auto update:</span>' +
-            '    <span class="u-h">is</span>' +
-            '    <span class="rounded-icon live-toggle__value">Off</span>' +
-            '    <span class="u-h">(turn on)</span>' +
-            '  </button>';
-
-        this.view = {
-            render: function (res) {
-                var $attachTo = [bonzo(options.attachTo[0]), bonzo(options.attachTo[1])],
-                    date = new Date().toString(),
-                    resultHtml = [
-                        $.create('<div>' + res[options.responseField[0]] + '</div>')[0],
-                        $.create('<div>' + res[options.responseField[1]] + '</div>')[0]
-                    ],
-                    elementsToAdd = toArray(resultHtml[0].children);
-
-                this.unreadBlocks += resultHtml[0].children.length;
-
-                bonzo(resultHtml[0].children).addClass('autoupdate--hidden');
-                $attachTo[0].prepend(elementsToAdd);
-                $attachTo[1].prepend(toArray(resultHtml[1].children));
-
-                if (elementsToAdd.length) {
-                    mediator.emit('modules:autoupdate:updates', elementsToAdd.length);
-                }
-                // add a timestamp to the attacher
-                $attachTo[0].attr('data-last-updated', date);
-                twitter.enhanceTweets();
-
-                if (this.isUpdating && detect.pageVisible()) {
-                    this.notificationBar.setState('hidden');
-                    this.view.revealNewElements.call(this);
-                } else if (this.unreadBlocks > 0) {
-                    this.notificationBar.notify(this.unreadBlocks);
-                    mediator.emit('modules:autoupdate:unread', this.unreadBlocks);
+        var scrolledPastTopBlock = function () {
+            return $liveblogBody.offset().top < window.pageYOffset;
+        };
+        var isLivePage = window.location.search.indexOf('?page=') === -1;
+
+
+        var setUpListeners = function () {
+            bean.on(document.body, 'click', '.toast__button', function () {
```

Comment:
Are `.toast__button`s injected in the page? If not then you can save a query either by attaching the listener on the button itself (here it will fire every time there's a click) or (I learned this while reading bean.js source) by passing it instead:

`bean(document.body, 'click', [$toastButton[0]], ...)`